examples/llava/clip.cpp

```diff
@@ -2475,6 +2553,12 @@ bool clip_image_batch_encode(clip_ctx * ctx, const int n_threads, const clip_ima
     if (ctx->has_minicpmv_projector) {
         GGML_ASSERT(batch_size == 1);
     }
+    if(ctx->has_glm_projector){
+        GGML_ASSERT(batch_size == 1);
+        ggml_tensor * boi = ctx->vision_model.boi_w;
```

Comment:
I don't quite get how these `boi` and `eoi` are used. I guess that they are 1d array of the begin-of-image and end-of-image token embedding, right?

If that's incorrect, you explain this a bit further? Thanks.

Reply:
An alternative way is to use `ggml_concat` inside `clip_image_build_graph` to "glue" these tokens with patch embeddings. But what you're doing is also acceptable, I'll optimize it more after the refactoring.